numpy/lib/function_base.py

```diff
@@ -175,54 +175,102 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
             raise AttributeError(
                 'max must be larger than min in range parameter.')
 
+    # Histogram is an integer or a float array depending on the weights.
+    if weights is None:
+        ntype = int
```

Comment:
Ah ok - did you mean to comment on the line with ``indices = tmp_a.astype(int)``?